driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -151,10 +151,46 @@ public void onSuccess(List<Void> l) {
                     initFuture.set(null);
                 }
             }
+
+            @Override
+            public void onFailure(Throwable t) {
```

Comment:
I pushed a branch ([java822sc](https://github.com/datastax/java-driver/tree/java822sc), tests still in progress) which has some tests that demonstrate this.  The pool is closed and never added to SessionManagerPools so some tests NPE, particularly [HostConnectionPoolTest#should_not_mark_host_down_if_some_connections_fail_on_init()](https://github.com/datastax/java-driver/blob/java822sc/driver-core/src/test/java/com/datastax/driver/core/HostConnectionPoolTest.java#L681-L737).